src/java/org/apache/cassandra/schema/IndexMetadata.java

```diff
@@ -106,20 +113,53 @@ public static IndexMetadata fromIndexTargets(List<IndexTarget> targets,
         return new IndexMetadata(name, newOptions, kind);
     }
 
-    public static String generateDefaultIndexName(String table, ColumnIdentifier column)
+    /**
+     * Generates a default index name from the table and column names.
+     * Characters other than alphanumeric and underscore are removed.
+     * Long index names are truncated to fit the length allowing constructing filenames.
+     *
+     * @param table  the table name
+     * @param column the column identifier. Can be null if the index is for the primary key.
```

Comment:
I think it's null if the index is for the entire table, as in certain custom implementations (DSE search and some 3rd parties).

Reply:
Improved the comment to not speculate on the null case.